### PR TITLE
[r] Push `schema` accessor down to `libtiledbsoma`

### PR DESCRIPTION
**Pull request by johnkerl**
**Issue and/or context:** #2406 #3055 [[sc-55682]](https://app.shortcut.com/tiledb-inc/story/55682/r-use-libtiledbsoma-for-all-tiledb-schema-callsites)

**Changes:**

**Notes for Reviewer:**

### CODE REVIEW

**Comment by johnkerl** [comment]
Update as of 2024-09-30:

* Fails unit test [here](https://github.com/single-cell-data/TileDB-SOMA/blob/40e21784e953d808dbf1b78b2f9ede5ad185be0e/apis/r/tests/testthat/test-SOMADataFrame.R#L641-L642)
  * Old/new schema comparison fails in `check_arrow_schema_data_types` in `R/utils-arrow.R`: value-type binary vs value-type string
* I've traced this down to this [one line](https://github.com/single-cell-data/TileDB-SOMA/blob/40e21784e953d808dbf1b78b2f9ede5ad185be0e/apis/r/R/SOMADataFrame.R#L339)
  * TileDB-R's schema evolution (not yet ported to libtiledbsoma) writes value-type binary
  * Later on in the unit-test spot mentioned above that fails to check against value-type readback using the schema getter on this PR (which has been ported to libtiledbsoma)

File: apis/r/R/SOMADataFrame.R [hunk 1]
```diff
@@ -245,6 +245,10 @@ SOMADataFrame <- R6::R6Class(
           is.data.frame(values) || is_arrow_table(values) || is_arrow_record_batch(values)
       )
 
+      # Leave state unmodified
+      # TODO: this issue will automatically go away on https://github.com/single-cell-data/TileDB-SOMA/issues/3059
+      on.exit(self$reopen(mode = "WRITE"))
```
Comment (mojaveazure) [inline apis/r/R/SOMADataFrame.R:250]: If `mode` ever expands beyond `"READ"` and `"WRITE"`, or changes to `"r"`/`"w"` to align with Python, change to capture the array mode and reopen in said mode
```suggestion
      omode <- self$mode()
      on.exit(self$reopen(mode = omode))
```

**Review by mojaveazure — APPROVED** [review]
LGTM; left one minor comment to future-proof the code. Please remember to bump the develop version, update the changelog, and 🚢

File: apis/r/R/SOMADataFrame.R (hunk 1 above)
Comment (johnkerl) [reply apis/r/R/SOMADataFrame.R]: Awesome, thanks @mojaveazure !! :)

**Pull request merged**